e2e/LongHaul/amqpPooling/DeviceManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Mash.Logging;
+
+namespace Microsoft.Azure.Devices.LongHaul.AmqpPooling
+{
+    internal class DeviceManager : IAsyncDisposable
+    {
+        private readonly Logger _logger;
+        private readonly int _devicesCount;
+
+        private const string DevicePrefix = "LongHaulAmqpPoolingDevice_";
+
+        private static IotHubServiceClient s_serviceClient;
+        private static IList<Device> s_devices;
+
+        public DeviceManager(Logger logger, Parameters parameters)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _devicesCount = parameters.DevicesCount;
+
+            s_serviceClient = new IotHubServiceClient(parameters.IotHubConnectionString);
+            s_devices = new List<Device>(_devicesCount);
+        }
+
+        public async Task<IList<Device>> AddDevicesAsync(CancellationToken ct)
+        {
+            _logger.Trace($"Start creating totally {_devicesCount} devices.", TraceSeverity.Information);
+
+            for (int i=0; i< _devicesCount; i++)
+            {
+                string deviceId = DevicePrefix + i;
+                _logger.Trace($"Creating a device with Id {deviceId}", TraceSeverity.Verbose);
+
+                var device = new Device(deviceId);
+                device = await s_serviceClient.Devices.CreateAsync(device, ct).ConfigureAwait(false);
+                s_devices.Add(device);
+            }
+
+            return s_devices;
```

Comment:
Might this work? Doesn't include logging, though, but you can add that in by changing the select statement to use a code block.

return _devicesCount
    .Select(id => s_serviceClient.Devices.CreateAsync(new Device(DevicePrefix + id))
    .ToList();

Reply:
No, it's not working. I didn't find it simple to use LINQ here, so I would rather keep it as-is.